src/modules/vendor/components/courses/CourseCreationModal.vue

```diff
@@ -96,10 +100,14 @@ export default {
   watch: {
     'newCourse.program': function (value) {
       const selectedProgram = this.programs.find(p => p._id === value);
+      const filteredSubPrograms = selectedProgram
+        ? selectedProgram.subPrograms
+          .filter(sp => !sp.isStrictlyELearning && sp.status === PUBLISHED)
+        : [];
 
-      if (get(selectedProgram, 'subPrograms.length')) {
+      if (filteredSubPrograms.length) {
```

Comment:
tu peux simplifier en 
```
this.subProgramOptions = formatAndSortOptions(filteredSubPrograms, 'name');
this.disableSubProgram = this.subProgramOptions.length;
if (!this.subProgramOptions.length) this.update('', 'subProgram');
else if (this.subProgramOptions.length === 1) this.update(this.subProgramOptions[0].value, 'subProgram');
```

Reply:
NB : si tu mets en place ma proposition du dessus, tu peux directement mettre 
`this.subProgramOptions = formatAndSortOptions(selectedProgram.blendedPublishedSubPrograms, 'name');`

Reply:
done